process/block/metablock.go

```diff
@@ -1603,75 +1624,52 @@ func (mp *metaProcessor) createEpochStartForMetablock(metaBlock *block.MetaBlock
 		lastNotarizedHeaders[i] = lastNotarizedHdr
 	}
 
-	pendingMiniBlocks, err := mp.pendingMiniBlocks.PendingMiniBlockHeaders(lastNotarizedHeaders)
-	if err != nil {
-		return nil, err
-	}
-
-	for _, pendingMiniBlock := range pendingMiniBlocks {
-		recvShId := pendingMiniBlock.ReceiverShardID
-
-		currShardPendingMBs := epochStart.LastFinalizedHeaders[recvShId].PendingMiniBlockHeaders
-		currShardPendingMBs = append(currShardPendingMBs, pendingMiniBlock)
-	}
-
-	return epochStart, nil
+	return epochStart, lastNotarizedHeaders, nil
 }
 
 func (mp *metaProcessor) getLastFinalizedMetaHashForShard(shardHdr *block.Header) ([]byte, []byte, error) {
 	var lastMetaHash []byte
 	var lastFinalizedMetaHash []byte
-
-	numAddedMetas := len(shardHdr.MetaBlockHashes)
-	if numAddedMetas > 0 {
-		lastMetaHash = shardHdr.MetaBlockHashes[numAddedMetas-1]
-	}
-
-	if numAddedMetas > 1 {
-		lastFinalizedMetaHash = shardHdr.MetaBlockHashes[numAddedMetas-2]
-		return lastMetaHash, lastFinalizedMetaHash, nil
-	}
+	var err error
 
 	for currentHdr := shardHdr; currentHdr.GetNonce() > 0 && currentHdr.GetEpoch() == shardHdr.GetEpoch(); {
-		oldShardHdr, err := process.GetShardHeader(currentHdr.GetPrevHash(), mp.dataPool.ShardHeaders(), mp.marshalizer, mp.store)
-		if err != nil {
-			return nil, nil, err
-		}
-
-		if len(oldShardHdr.MetaBlockHashes) == 0 {
-			currentHdr = oldShardHdr
+		if len(currentHdr.MetaBlockHashes) == 0 {
+			currentHdr = currentHdr
```

Comment:
There is something wrong here, as you renamed oldShardHdr with currentHdr. Check whole this method again

Reply:
done